test/test_mps.py

```diff
@@ -10586,6 +10586,7 @@ class TestAdvancedIndexing(TestCaseMPS):
     supported_dtypes = [torch.float32, torch.float16, torch.int64, torch.int32, torch.int16, torch.uint8]
     supported_np_dtypes = [np.float32, np.float16, np.int64, np.int32, np.int16, np.uint8]
 
+    @unittest.skipIf(product_version < 14.0, "Skipped on macOS < 14")
```

Comment:
Alas, such tests are super unreliable as we only warn once. Is there a way to reset warnings?

Reply:
I'm not sure if there is a way to do so. That being said, this test is ok because it checks no warning emitted on macOS 14+, and indeed, if our fallback condition is right, there shouldn't be any warning emitted.